plio/views.py

```diff
@@ -215,6 +237,105 @@ def duplicate(self, request, uuid):
         plio.save()
         return Response(self.get_serializer(plio).data)
 
+    @action(
+        methods=["post"],
+        detail=True,
+        permission_classes=[IsAuthenticated, PlioPermission],
+    )
+    def copy(self, request, uuid):
+        """copies the given plio to another workspace"""
+        for key in ["workspace"]:
+            if key not in request.data:
+                return Response(
+                    {"detail": f"{key} is not provided"},
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+        # return 404 if user cannot access the object
+        # else fetch the object
+        plio = self.get_object()
+
+        if plio.video is not None:
+            video = Video.objects.filter(id=plio.video.id).first()
+            video.pk = None
+
+            items = list(Item.objects.filter(plio__id=plio.id))
+            questions = list(Question.objects.filter(item__plio__id=plio.id))
+
+            # will be needed to handle questions with images
+            question_indices_with_image = []
+            images = []
+
+            for index, question in enumerate(questions):
+                if question.image is not None:
+                    question_indices_with_image.append(index)
+                    images.append(question.image)
+        else:
+            video = None
+            items = []
+            questions = []
+
+        # django will auto-generate the key when the key is set to None
+        plio.pk = None
+        plio.uuid = None
+        plio.status = "draft"
+
+        # change workspace
+        workspace = request.data.get("workspace")
+        success = set_tenant(workspace)
+
+        if not success:
+            return Response(
+                {"detail": "workspace does not exist"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
```

Comment:
This should be checked in the beginning no?
If the workspace doesn't exist, then no need to do any of the steps written above